vita3k/shader/src/translator/ialu.cpp

```diff
@@ -121,10 +115,27 @@ bool USSETranslatorVisitor::vbw(
     case Opcode::SHR: operation = spv::Op::OpShiftRightLogical; break;
     default: return false;
     }
-
-    result = m_b.createBinOp(operation, type_ui32, src1, src2);
-    if (m_b.isFloatType(m_b.getTypeId(src2))) {
-        result = m_b.createUnaryOp(spv::Op::OpBitcast, type_f32, src2);
+    //optimisation. (any OR 0 || any XOR 0 || any AND 0xFFFFFFFF) -> assign
+    bool is_no_op = (operation == spv::Op::OpBitwiseOr || operation == spv::Op::OpBitwiseXor) && (m_b.getOpCode(src2) == spv::Op::OpConstant) && (m_b.getConstantScalar(src2) == 0);
+    is_no_op = is_no_op || (operation == spv::Op::OpBitwiseAnd) && (m_b.getOpCode(src2) == spv::Op::OpConstant) && (m_b.getConstantScalar(src2) == 0xFFFFFFFF);
```

Comment:
sorry bad view, but code here is strange 🤔 , have code in double no understund

Reply:
X OR 0 = X
X XOR 0 = X
X AND MAX_32BIT (0xFFFFFFFF) == X

It's like that

Reply:
> X OR 0 = X X XOR 0 = X X AND MAX_32BIT (0xFFFFFFFF) == X
> 
> It's like that

no possible set all in one definition value ? 🤔

Reply:
Yes possible. Need multiple lines though. I don't require. If you want you can request.

Reply:
> Yes possible. Need multiple lines though. I don't require. If you want you can request.

is like you want you, iss shader stuff, i just said recomandation for more readeable code, i have thinks also can create 2 value and check both :)